reactive/cinder_storpool_charm.py

```diff
@@ -87,6 +93,195 @@ def configure():
     reactive.set_state('cinder-storpool.configured')
 
 
+@reactive.hook('leader-elected')
+def we_are_the_leader():
+    """
+    Make note of the fact that this unit has been elected as the leader for
+    the `cinder-storpool` charm.  This will prompt the unit to send presence
+    information to the other charms along the `storpool-presence` hook.
+    """
+    rdebug('have we really been elected leader?')
```

Comment:
This code down to the `try:` is not really needed.  If this unit is the leader, then `leader_set(...)` will succeed, otherwise it will fail.  Thus, checking `is_leader()` isn't useful if the next thing the charm is going to do is to try to set leadership.

`is_leader(...)` is provided to avoid doing a lot of processing and *then* trying to set leadership (which could then fail).  It's a shortcut to avoid doing too much work in a hook.